File: LeetCode/453. Minimum Moves to Equal Array Elements/Solution.py
class Solution:

    def minMoves(self, nums) -> int:
        nums.sort()
        count = 0 
        min_val = nums[0]
        for i in range(len(nums)-1, 0, -1):
            val = nums[i] + count
            difference = val - min_val
            count+=difference
            min_val = val
        return count

File: LeetCode/453. Minimum Moves to Equal Array Elements/Solution.java
class Solution {
    public int minMoves(int[] nums) {
        // find max of the nums
        int min = Integer.MAX_VALUE, sum = 0, n = nums.length;
        for (int num : nums)
        {
            min = Math.min(num, min);
            sum += num;
        }
        
        return sum - n*min;
    }
}